### PR TITLE
Issue 490 reformat lead acid

### LINKED ISSUE #490: Update lead-acid functionality in new model format
- [x] Side reactions
- [x] Surface form
- [ ] First-order composite current form
- [x] Remove empty tests

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Fixes surface forms for #490 
I have skipped the 1+1D tests for now, to avoid clashes with #477 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pybamm/models/submodels/base_submodel.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def __init__(self, param, domain=None):
         self.events = {}
 
         self.domain = domain
+        self.set_domain_for_broadcast()
```
Comment (rtimms) [inline pybamm/models/submodels/base_submodel.py]: handy!

File: pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/full_surface_form_stefan_maxwell_conductivity.py [hunk 2]
```diff
@@ -1,11 +1,234 @@
 #
```
Comment (rtimms) [inline pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/full_surface_form_stefan_maxwell_conductivity.py]: Should this have been replace by lines 4-6?

File: pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py [hunk 3]
```diff
@@ -1,11 +1,120 @@
 #
-# Class for leading-order surface form electrolyte conductivity employing stefan-maxwell
+# Class for full surface form electrolyte conductivity employing stefan-maxwell
```
Comment (rtimms) [inline pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py]: full or leading-order here?

**Comment by Scottmar93** [comment]
Looking good, ill review properly when in tomorrow morning :)

**Comment by tinosulzer** [comment]
haha ok I'll be less trigger happy with the merge than last time dw

File: pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py [hunk 4]
```diff
@@ -1,11 +1,120 @@
...
+        delta_phi = variables[
+            "Average " + self.domain.lower() + " electrode surface potential difference"
+        ]
+
+        if self.domain == "Negative":
+            C_dl = param.C_dl_n
+        elif self.domain == "Positive":
+            C_dl = param.C_dl_p
+
+        if isinstance(j, pybamm.Broadcast):
```
Comment (Scottmar93) [inline pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py]: If j is a broadcast then `j_av = pybamm.average(j) = j.orphans[0]` ?

File: pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py [hunk 5]
```diff
@@ -16,21 +125,25 @@ class LeadingOrder(BaseLeadingOrder):
         The parameters to use for this submodel
 
 
-    **Extends:** :class:`pybamm.electrolyte.stefan_maxwell.conductivity.surface_potential_form.BaseLeadingOrder`
-    """  # noqa: E501
+    **Extends:** :class:`BaseLeadingOrderSurfaceForm`
```
[the remark this answers is not in the document]
Comment (tinosulzer) [reply pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py]: Will check after merging

File: pybamm/models/submodels/porosity/leading_reaction_driven_porosity.py [hunk 6]
```diff
@@ -28,15 +28,15 @@ def get_fundamental_variables(self):
 
     def get_coupled_variables(self, variables):
 
-        j_n_av = variables["Average negative electrode interfacial current density"]
-        j_p_av = variables["Average positive electrode interfacial current density"]
+        j_n = variables["Negative electrode interfacial current density"].orphans[0]
```
[the remark this answers is not in the document]
Comment (tinosulzer) [reply pybamm/models/submodels/porosity/leading_reaction_driven_porosity.py]: Opened #516 for this

File: pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py (hunk 4 above)
Comment (tinosulzer) [reply pybamm/models/submodels/electrolyte/stefan_maxwell/conductivity/surface_potential_form/leading_surface_form_stefan_maxwell_conductivity.py]: Opened #516 for this

**Pull request merged**